src/main/scala/org/apache/spark/sql/execution/datasources/greenplum/GreenplumUtils.scala

```diff
@@ -299,3 +304,21 @@ object GreenplumUtils extends Logging {
     }.getOrElse(df)
   }
 }
+
+private[greenplum] case class CanonicalTblName(schema: Option[String], rawName: Option[String])
+
+/**
+ * Extract schema name and raw table name from a table name string.
+ */
+private[greenplum] object TableNameExtractor {
+  private val nonSchemaTable = """[\"]*([0-9a-zA-Z_]+)[\"]*""".r
+  private val schemaTable = """([\"]*[0-9a-zA-Z_]+[\"]*)\.[\"]*([0-9a-zA-Z_]+)[\"]*""".r
+
+  def extract(tableName: String): CanonicalTblName = {
+    tableName match {
+      case nonSchemaTable(table) => CanonicalTblName(None, Some(table))
+      case schemaTable(schme, table) => CanonicalTblName(Some(schme), Some(table))
+      case _ => throw new IllegalArgumentException("The table name is illegal.")
```

Comment:
how about notify user the illegal table name and how to collect

Reply:
done.